MySQL added a new connection option MYSQL_ENABLE_CLEARTEXT_PLUGIN and believes
that it is there in every version > 5.5.27. However, it is not in mariadb-5.5.31,
so we cannot use it.

MariaDB upstream report: https://mariadb.atlassian.net/browse/MDEV-4573 

diff -up mysql-connector-odbc-5.2.5-src/driver/connect.c.cleartext mysql-connector-odbc-5.2.5-src/driver/connect.c
--- mysql-connector-odbc-5.2.5-src/driver/connect.c.cleartext	2013-05-24 15:57:18.954565190 +0200
+++ mysql-connector-odbc-5.2.5-src/driver/connect.c	2013-05-24 15:58:03.194564417 +0200
@@ -257,7 +257,7 @@ SQLRETURN myodbc_do_connect(DBC *dbc, Da
   }
 #endif
 
-#if (MYSQL_VERSION_ID >= 50527 && MYSQL_VERSION_ID < 50600) || MYSQL_VERSION_ID >= 50607
+#if ((MYSQL_VERSION_ID >= 50527 && MYSQL_VERSION_ID < 50600) || MYSQL_VERSION_ID >= 50607) && defined (MYSQL_ENABLE_CLEARTEXT_PLUGIN)
   if (ds->enable_cleartext_plugin)
   {
     mysql_options(mysql, MYSQL_ENABLE_CLEARTEXT_PLUGIN, (char *)&on);
